# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/MemoryMetrics.cpp
# Commit: 6c762044624f
# Full Hash: 6c762044624f7d9c77b343238b7f9ff9979a5888
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2021-01-21 03:22:45
# Regressor Bug: 1657025
# File Overlap Count: 1
# Description:
#   Bug 1657025 - Wait for all helper threads to finish before collecting memory stats r=sfink
#   
#   Try harder to get this right.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D102391
# ==============================================================================

diff -r abdfa91676ff -r 6c762044624f js/src/vm/MemoryMetrics.cpp
--- a/js/src/vm/MemoryMetrics.cpp	Tue Jan 19 15:20:25 2021 +0000
+++ b/js/src/vm/MemoryMetrics.cpp	Wed Jan 20 17:52:08 2021 +0000
@@ -618,9 +618,11 @@
   // Finish any ongoing incremental GC that may change the data we're gathering
   // and ensure that we don't do anything that could start another one.
   gc::FinishGC(cx);
-  gc::WaitForBackgroundTasks(cx);
   JS::AutoAssertNoGC nogc(cx);
 
+  // Wait for any background tasks to finish.
+  WaitForAllHelperThreads();
+
   if (!rtStats->realmStatsVector.reserve(rt->numRealms)) {
     return false;
   }
